code/obj/machinery/drone_beacon.dm

```diff
@@ -0,0 +1,80 @@
+#define PLAYER_SEEK_RANGE 12
+#define DRONE_SEEK_RANGE 25
+
+/obj/machinery/drone_beacon
+	name = "Mysterious Beacon"
+	desc = "Some strange transmitter."
+	icon = 'icons/obj/ship.dmi'
+	icon_state = "beacon_synd"
+	anchored = TRUE
+	density = TRUE
+	processing_tier = PROCESSING_EIGHTH
+	_max_health = 500
+	_health = 500
+
+	New()
+		..()
+		src.AddComponent(/datum/component/obj_projectile_damage)
+		RegisterSignal(src, COMSIG_ATOM_HITBY_PROJ, .proc/hitby_proj)
+
+	ex_act(severity)
+		summon_drones()
+		..()
+
+	attackby(obj/item/I, mob/user)
+		summon_drones()
+		..()
+
+	disposing()
+		UnregisterSignal(src, COMSIG_ATOM_HITBY_PROJ)
+		..()
+
+	process()
+		var/list/target_list = list()
+		for(var/mob/M in range(PLAYER_SEEK_RANGE, src))
+			if(M.stat == 2) //this feels like there should be a define for .stat
+				continue
+			if(istype(M.loc, /obj/machinery/vehicle))
+				continue
+			target_list += M
+		if(!length(target_list))
+			for(var/obj/machinery/vehicle/V in range(PLAYER_SEEK_RANGE, src))
```

Comment:
Once again makes me nervous, iterating through `tr_cat[TR_CAT_PODS_AND_CRUISERS]` or something similar might be better.